### PR TITLE
sia agents - dynamic role cert

### PR DESCRIPTION
**Pull request by OferLevi85**
Signed-off-by: Ofer Levi <ofer.levi@verizonmedia.com>

### CODE REVIEW

File: libs/go/sia/aws/agent/agent.go [hunk 1]
```diff
@@ -91,11 +91,35 @@ func GetRoleCertificate(ztsUrl, svcKeyFile, svcCertFile string, opts *options.Op
 		return false
 	}
 
-	//initialize our return state to success
-	failures := 0
+	//fetch static roles
+	failures := fetchRoles(opts.Roles, opts, key, svcKeyFile, client)
```
Comment (havetisyan) [inline libs/go/sia/aws/agent/agent.go:95]: I'm somewhat confused with the lines 95-101. 

1. In the config we claim that //if true, fetch role certs dynamically. Roles attribute will be ignored. Based on the comments and this code we're fetching static roles and then dynamic roles.

2. when we address the fetch dynamic role part - why aren't we checking if the option is enabled in the first place? it looks like we always fetch the dynamic role certs.

File: provider/azure/sia-vm/authn.go [hunk 2]
```diff
@@ -77,12 +78,40 @@ func GetRoleCertificate(ztsUrl, svcKeyFile, svcCertFile string, opts *options.Op
 		return false
 	}
 
-	// initialize our return state to success
-	failures := 0
+	// fetch static roles
+	failures:= fetchRoles(opts.Roles, opts, key, provider, client)
```
Comment (havetisyan) [inline provider/azure/sia-vm/authn.go:82]: same comments as the one for aws above.

**Review by havetisyan — CHANGES_REQUESTED** [review]

File: libs/go/sia/aws/agent/agent.go (hunk 1 above)
Comment (OferLevi85) [reply libs/go/sia/aws/agent/agent.go]: Fixed

File: provider/azure/sia-vm/authn.go (hunk 2 above)
Comment (OferLevi85) [reply provider/azure/sia-vm/authn.go]: Fixed

**Pull request closed without merging**